arch/xtensa/src/common/xtensa_int_handlers.S

```diff
@@ -263,6 +286,19 @@ _xtensa_level1_handler:
 	mov		a2, sp							      /* Address of state save on stack */
 	call0	_xtensa_context_save			/* Save full register state */
 
+  /* Save current SP before (possibly) overwriting it, it's the register save
+   * area. This value will be used later by dispatch_c_isr to retrive the
+   * register save area.
+   */
+
+  mov  a12, sp
+
+  /* Switch to an interrupt stack if we have one */
+
+#if CONFIG_ARCH_INTERRUPTSTACK > 15
+  setintstack a13 a14
+#endif
+
```

Comment:
I think we need to adjust (i.e. subtract 16?) the stack pointer (a1?) to avoid overlap, because g_cpu_intstack_top does not include offset now.

Reply:
I'll run more tests with the adjustment this evening and verify the memories.  Thanks for the remainder.
(Yes, a1 is sp)

Reply:
@masayuki2009 I don't think that's necessary, same as the discussion we were having with ARM, the SP is decremented on procedures entry.

Reply:
@Ouss4 
Sorry for confusing you.

Reply:
No problem, there was no confusion :) 
I did look here and there to see how others are doing the same thing and I found it similar to what's proposed by this PR.